cylc/flow/ws_data_mgr.py

```diff
@@ -356,18 +358,16 @@ def generate_ghost_families(self, family_proxies=None, cycle_points=None):
             # first-parent single-inheritance tree
             if not cycle_points or point_string not in cycle_points:
                 continue
-            cycle_first_parents = set()
+            cycle_first_parents = []
```

Comment:
Are we going to have duplicate first parents? Is that why we are replacing the list by a set now?

Reply:
Hmm... Actually yes, possibly, I wanted to preserve order.. Hold on will change it

Reply:
I don't think order actually matters for this XD.. my bad.